### ISSUE TITLE
UIImage(GIF) sd_animatedGIFWithData crash

### ISSUE THREAD

**Issue by raomengyun** [rs/SDWebImage#1837]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

````
#15 Thread
NSInvalidArgumentException
*** -[__NSPlaceholderArray initWithObjects:count:]: attempt to insert nil object from objects[0]
0 CoreFoundation	___exceptionPreprocess + 124
1 libobjc.A.dylib	objc_exception_throw + 56
2 CoreFoundation	-[__NSPlaceholderArray initWithObjects:count:] + 312
3 CoreFoundation	+[NSArray arrayWithObjects:count:] + 52
4 SDWebImage	+[UIImage(GIF) sd_animatedGIFWithData:] + 300
5 SDWebImage	+[UIImage(MultiFormat) sd_imageWithData:] + 100
6 SDWebImage	-[SDWebImageDownloaderOperation URLSession:task:didCompleteWithError:] + 632
7 SDWebImage	-[SDWebImageDownloader URLSession:task:didCompleteWithError:] + 128
8 CFNetwork	___51-[NSURLSession delegate_task:didCompleteWithError:]_block_invoke.185 + 72
9 Foundation	___NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
10 Foundation	-[NSBlockOperation main] + 96
11 Foundation	-[__NSOperationInternal _start:] + 620
12 Foundation	___NSOQSchedule_f + 228
13 libdispatch.dylib	__dispatch_client_callout + 16
14 libdispatch.dylib	__dispatch_queue_serial_drain + 928
15 libdispatch.dylib	__dispatch_queue_invoke + 652
16 libdispatch.dylib	__dispatch_root_queue_drain + 572
17 libdispatch.dylib	__dispatch_worker_thread3 + 124
18 libsystem_pthread.dylib	_pthread_wqthread + 1288
19 libsystem_pthread.dylib	start_wqthread + 4
````

We found this code in `UIImage+GIF.m`
````
CGImageRef CGImage = CGImageSourceCreateImageAtIndex(source, 0, NULL);
#if SD_UIKIT || SD_WATCH
        UIImage *frameImage = [UIImage imageWithCGImage:CGImage scale:scale orientation:UIImageOrientationUp];
        staticImage = [UIImage animatedImageWithImages:@[frameImage] duration:0.0f];
#elif SD_MAC
        staticImage = [[UIImage alloc] initWithCGImage:CGImage size:NSZeroSize];
#endif
        CGImageRelease(CGImage);
````

The document of `+ (UIImage *)imageWithCGImage:(CGImageRef)cgImage scale:(CGFloat)scale orientation:(UIImageOrientation)orientation;`say that the method will return a new image object for the specified Quartz image, or **nil** if the method could not initialize the image from the specified image reference.

**Comment by mythodeia** [comment]
can you replicate this with a specific image in the demo project?

**Comment by mythodeia** [comment]
closing this. please reopen with more info

**Comment by dreampiggy** [comment]
Crash Log: (SDWebImge 4.1)

Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x1882c2fe0 __exceptionPreprocess
1  libobjc.A.dylib                0x186d24538 objc_exception_throw
2  CoreFoundation                 0x1881a42d8 -[__NSPlaceholderArray initWithObjects:count:]
3  CoreFoundation                 0x1881b0030 +[NSArray arrayWithObjects:count:]
4  News                           0x1012e90dc +[UIImage(GIF) sd_animatedGIFWithData:] (UIImage+GIF.m:48)
5  News                           0x1012e9210 +[UIImage(MultiFormat) sd_imageWithData:] (UIImage+MultiFormat.m:28)
6  News                           0x1012e3f50 -[SDWebImageDownloaderOperation URLSession:task:didCompleteWithError:] (SDWebImageDownloaderOperation.m:428)
7  News                           0x1012e1704 -[SDWebImageDownloader URLSession:task:didCompleteWithError:] (SDWebImageDownloader.m:324)
8  CFNetwork                      0x188a9bad4 __51-[NSURLSession delegate_task:didCompleteWithError:]_block_invoke.185
9  Foundation                     0x188d92854 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__
10 Foundation                     0x188cd77b0 -[NSBlockOperation main]
11 Foundation                     0x188cc7b68 -[__NSOperationInternal _start:]
12 Foundation                     0x188d94bf0 __NSOQSchedule_f
13 libdispatch.dylib              0x18717a9a0 _dispatch_client_callout
14 libdispatch.dylib              0x187188ad4 _dispatch_queue_serial_drain
15 libdispatch.dylib              0x18717e2cc _dispatch_queue_invoke
16 libdispatch.dylib              0x18718aa50 _dispatch_root_queue_drain
17 libdispatch.dylib              0x18718a7d0 _dispatch_worker_thread3
18 libsystem_pthread.dylib        0x187383100 _pthread_wqthread
19 libsystem_pthread.dylib        0x187382cac start_wqthread

**Comment by dreampiggy** [comment]
Fixed via #1991
